src/api/BaseServiceApi.ts

```diff
@@ -0,0 +1,54 @@
+import { FetchMethods, IBaseServiceApi, ServiceResponseType } from '@/types/serviceTypes'
+import { BaseErrorInstance } from './BaseError'
+import { Data } from '@/types/dataTypes'
+
+export class BaseServiceApi implements IBaseServiceApi {
+  private async handleFetch(
+    method: FetchMethods,
+    url: string,
+    defaultMessage: string,
+    body?: Data
+  ) {
+    try {
+      const response = await fetch(url, {
+        method: method,
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: body ? JSON.stringify(body) : undefined,
+      })
+      if (!response.ok) {
+        throw Error(`${response.status}: ${response.statusText}`)
+      }
+      const data: ServiceResponseType<typeof method> = await response.json()
+      return data
+    } catch (error) {
+      BaseErrorInstance.throwError(error, defaultMessage)
```

Comment:
throw new this.error(Failed during json parsing,  ${response.status}, ${response.statusText} )